contracts/contracts/solidity/StarknetValidator.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >0.6.0 <=0.8.0;
+
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorValidatorInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/TypeAndVersionInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AccessControllerInterface.sol';
+import '@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol';
+import '@chainlink/contracts/src/v0.8/SimpleWriteAccessController.sol';
+
+import '@chainlink/contracts/src/v0.8/dev/vendor/openzeppelin-solidity/v4.3.1/contracts/utils/Address.sol';
+
+import './interfaces/IStarknetCore.sol';
+import './utils/utils.sol';
+
+/**
+ * @title Validator - makes cross chain call to update the Sequencer Uptime Feed on L2
+ */
+contract Validator is TypeAndVersionInterface, AggregatorValidatorInterface, SimpleWriteAccessController {
+    int256 private constant ANSWER_SEQ_OFFLINE = 1;
+    /* Selector hardcoded because StarkNet generates selectors differently than the standard ethereum way
+    different hash function on stark curve is used */
+    uint256 constant STARK_SELECTOR_UPDATE_STATUS =
+        1585322027166395525705364165097050997465692350398750944680096081848180365267;
+
+    IStarknetCore public immutable STARKNET_CORE;
+    uint256 public L2_UPTIME_FEED_ADDR;
+
+    /**
+     * @param starknetCore the address of the StarknetCore contract address
+     * @param l2UptimeFeedAddr the address of the Sequencer Uptime Feed on L2
+     */
+    constructor(address starknetCore, uint256 l2UptimeFeedAddr) {
+        require(starknetCore != address(0), 'Invalid Starknet Core address');
```

Comment:
This should be renamed once you replace local `IStarknetCore` with `IStarknetMessaging` from the submodule.